hibernate-core/src/main/java/org/hibernate/usertype/DynamicParameterizedType.java

```diff
@@ -33,6 +35,8 @@ interface ParameterType {
 
 		Class<?> getReturnedClass();
 
+		Type getReturnedJavaType();
```

Comment:
We didn't document that this type is for consumption only, so be safe, we should IMO add the new method with a default implementation.

```suggestion
		default Type getReturnedJavaType() {
			return getReturnedClass();
		}
```